### PR TITLE
136 persist winners for irv contests

### PR DESCRIPTION
**Pull request by vteague**
This adds a new database table for storing the assertion generation summary - basically a winner (if it succeeded) and an error message (if it failed). The database table is stored, and then used subsequently when generating reports.

As a nice bonus, this allows us to remove the winner from the get assertions request.

### CODE REVIEW

File: src/main/java/au/org/democracydevelopers/raireservice/controller/AssertionController.java [hunk 1]
```diff
@@ -99,12 +99,13 @@ public ResponseEntity<GenerateAssertionsResponse> serve(@RequestBody GenerateAss
     logger.debug(String.format("%s Calling raire-java with assertion generation request.",prefix));
     RaireResultOrError solution = generateAssertionsService.generateAssertions(request);
```
Comment (michelleblom) [inline src/main/java/au/org/democracydevelopers/raireservice/controller/AssertionController.java:101]: If the persisting is done on line 102, then I think we need to change the comment on 105 as we have already saved them to the database.

File: src/main/java/au/org/democracydevelopers/raireservice/persistence/entity/GenerateAssertionsSummary.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  @Column(name = "version", updatable = false, nullable = false)
+  private long version;
+
+  /**
+   * Name of the contest.
+   */
+  @Column(name = "contest_name", unique = true, updatable = false, nullable = false)
+  private String contestName = "";
+
+  /**
```
Comment (michelleblom) [inline src/main/java/au/org/democracydevelopers/raireservice/persistence/entity/GenerateAssertionsSummary.java:71]: Do we mean to make the following attributes public given there are methods for updating them?

File: src/main/java/au/org/democracydevelopers/raireservice/service/GetAssertionsJsonService.java [hunk 3]
```diff
@@ -141,4 +145,41 @@ public RaireSolution getRaireSolution(GetAssertionsRequest request)
       throw new RaireServiceException(ex.getMessage(), RaireErrorCode.INTERNAL_ERROR);
     }
   }
+
+  /**
+   * Retrieve the GenerateAssertionsSummary record for this contest, and either return the winner's
+   * index in the candidate list (if it is present and valid) or throw an error with the appropriate
+   * error message (if either there is no record, or the record contains an error rather than a winner).
```
Comment (michelleblom) [inline src/main/java/au/org/democracydevelopers/raireservice/service/GetAssertionsJsonService.java:152]: Typo: GenerateAssetionsSummary

**Review by michelleblom — APPROVED** [review]

File: src/main/java/au/org/democracydevelopers/raireservice/persistence/entity/GenerateAssertionsSummary.java (hunk 2 above)
Comment (vteague) [reply src/main/java/au/org/democracydevelopers/raireservice/persistence/entity/GenerateAssertionsSummary.java:71]: They have to be read by the getAssertions services - it's really another example of where 'public final' is the right thing, but I don't think final is compatible with the database. You're right, though, they shouldn't be publicly writeable - I've now changed them to private with getters.

**Pull request merged**